packages/xstate-store/src/useSyncExternalStoreWithSelector.ts

```diff
@@ -0,0 +1,101 @@
+// Copied from https://github.com/facebook/react/blob/0a76aecbfa3d493c6b97998d0444f6505f5e5365/packages/use-sync-external-store/src/useSyncExternalStoreWithSelector.js
```

Comment:
this hurts my soul, I completely miss the point of copy-pasting code just to claim the package has no redundant dependencies. Now imagine 10 libraries doing the same. As a result, the final application will have *more* code and not less.

Reply:
If there's a way we can fix the existing code (cc @tkdodo) then let's do that. Otherwise, I don't mind reintroducing the dependency.

Reply:
I'm not sure there is something to fix. If you create a new object or array in your selector, the comparison method must account for that. Per default, we use `===` to compare, so it won't work.

It would need:

```
const count = useSelector(
  store,
  (s) => s.context.count.slice(),
  shallowEqual
);
```

Reply:
I wonder why it worked with `useSyncExternalStoreWithSelector` then?

Reply:
this is why https://github.com/statelyai/xstate/pull/5084 :)